cppcache/include/geode/DataInput.hpp

```diff
@@ -489,11 +487,12 @@ class APACHE_GEODE_EXPORT DataInput {
 
  protected:
   /** constructor given a pre-allocated byte array with size */
-  DataInput(const uint8_t* m_buffer, size_t len, const CacheImpl* cache)
+  DataInput(const uint8_t* m_buffer, size_t len, const CacheImpl* cache,
+            const std::string& poolName)
       : m_buf(m_buffer),
         m_bufHead(m_buffer),
         m_bufLength(len),
-        m_poolName(EMPTY_STRING),
+        m_poolName(poolName),
```

Comment:
it must be a very bizarre past led to an internal member needing setter/getter - this is much more intuitive...